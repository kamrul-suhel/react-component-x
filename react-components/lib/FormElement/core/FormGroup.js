'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});
exports.default = FormGroup;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function FormGroup(props) {
	var disabledClass = props.disabled ? ' is-disabled' : '';
	var validClass = function () {
		switch (props.valid) {
			case true:
				return ' form-group-valid';
			case false:
				return ' form-group-invalid';
			default:
				return '';
		}
	}();
	var validationRequiredClass = props.validation ? ' form-group-validation-required' : '';
	var wideClass = props.wide ? ' form-group-wide' : '';
	var focusedClass = props.withFloatingLabel && !!props.isFocused ? ' is-focused' : '';
	var canFloatClass = props.withFloatingLabel ? ' can-float' : '';
	var shouldFloatClass = props.withFloatingLabel && (!!props.value || !!props.isFocused) ? ' should-float' : '';
	var className = props.className ? ' ' + props.className : '';
	var extraClassName = props.extraClassName ? ' ' + props.extraClassName : '';

	return _react2.default.createElement(
		'div',
		{ className: 'form-group form-type-' + props.type + wideClass + focusedClass + canFloatClass + shouldFloatClass + extraClassName + validationRequiredClass + validClass + disabledClass + className },
		props.children
	);
}