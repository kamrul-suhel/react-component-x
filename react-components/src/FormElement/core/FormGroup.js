import React from 'react';

export default function FormGroup(props) {
	const disabledClass = props.disabled ? ' is-disabled' : '';
	const validClass = (() => {
		switch (props.valid) {
			case true:
				return ' form-group-valid';
			case false:
				return ' form-group-invalid';
			default:
				return '';
		}
	})();
	const validationRequiredClass = props.validation ? ' form-group-validation-required' : '';
	const wideClass = props.wide ? ' form-group-wide' : '';
	const focusedClass = (props.withFloatingLabel && !!props.isFocused) ? ' is-focused' : '';
	const canFloatClass = props.withFloatingLabel ? ' can-float' : '';
	const shouldFloatClass = (props.withFloatingLabel && (!!props.value || !!props.isFocused)) ? ' should-float' : '';
	const className = props.className ? ` ${props.className}` : '';
	const extraClassName = props.extraClassName ? ` ${props.extraClassName}` : '';

	return (
		<div className={`form-group form-type-${props.type}${wideClass}${focusedClass}${canFloatClass}${shouldFloatClass}${extraClassName}${validationRequiredClass}${validClass}${disabledClass}${className}`}>
			{props.children}
		</div>
	);
}
