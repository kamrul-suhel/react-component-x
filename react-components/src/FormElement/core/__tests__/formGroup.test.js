import React from 'react';
import renderer from 'react-test-renderer';
import FormGroup from '../FormGroup';

it('renders correctly', () => {
	const tree = renderer.create(<FormGroup type="checkbox" disabled valid validation wide className="custom-class-name" />).toJSON();
	expect(tree).toMatchSnapshot();
});
