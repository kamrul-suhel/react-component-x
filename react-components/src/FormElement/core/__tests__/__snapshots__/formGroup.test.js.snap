// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="form-group form-type-checkbox form-group-wide form-group-validation-required form-group-valid is-disabled custom-class-name"
/>
`;
